engine/servermaster/http.go

```diff
@@ -16,43 +16,64 @@ package servermaster
 import (
 	"net/http"
 	"net/http/pprof"
+	"strings"
 
-	"github.com/gin-gonic/gin"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/pingcap/failpoint"
-	swaggerFiles "github.com/swaggo/files"
-	ginSwagger "github.com/swaggo/gin-swagger"
-
+	"github.com/pingcap/tiflow/engine/pkg/openapi"
 	"github.com/pingcap/tiflow/engine/pkg/promutil"
 	"github.com/pingcap/tiflow/pkg/util"
-
-	// Used for OpenAPI online docs.
-	_ "github.com/pingcap/tiflow/engine/docs/swagger"
 )
 
+// JobAPIPrefix is the prefix of the job API.
+const JobAPIPrefix = "/api/v1/jobs/"
+
 // RegisterRoutes create a router for OpenAPI
-func RegisterRoutes(router *gin.Engine, openapi *OpenAPI) {
-	// online docs
-	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
+func RegisterRoutes(router *http.ServeMux, grpcMux *runtime.ServeMux, forwardJobAPI http.HandlerFunc) {
+	// Swagger UI
+	router.HandleFunc("/swagger", openapi.SwaggerUI)
+	router.HandleFunc("/swagger/v1/openapiv2.json", openapi.SwaggerAPIv1)
 
-	// Open API
-	RegisterOpenAPIRoutes(router, openapi)
+	// Job API
+	router.HandleFunc("/api/v1/", func(w http.ResponseWriter, r *http.Request) {
+		if shouldForwardJobAPI(r) {
+			forwardJobAPI(w, r)
+		} else {
+			grpcMux.ServeHTTP(w, r)
+		}
```

Comment:
Can you add more comments on this logic?

Reply:
Done.